TombEngine/Game/camera.cpp

```diff
@@ -1304,6 +1304,45 @@ void CalculateCamera(const CollisionInfo& coll)
 	}
 }
 
+bool IsDeathCamera ()
+{
+	// If player is alive, it's not a death camera.
+	if (LaraItem->HitPoints > 0)
+		return false;
+	
+	// If Lara is in a special death animation (from extra_anims) triggered by enemies.
+	if (LaraItem->Animation.AnimObjectID == ID_LARA_EXTRA_ANIMS)
+		return true;
+
+	// Teeth spikes death animation
+	if (LaraItem->Animation.AnimNumber == LA_SPIKE_DEATH)
+	{
+		Camera.pos.RoomNumber = LaraItem->RoomNumber;
```

Comment:
I am not sure if it's nice idea to modify camera data inside a function called `IsDeathCamera`. It is a bad practice to change global variables in a function which is supposed just to check for certain conditions.

Reply:
Yeap, I took those commands away from the camera code in this next commit: [56665cd](https://github.com/MontyTRC89/TombEngine/pull/1449/commits/56665cd316a854d9c013ad4ac33f77308a7bd001)

Now they will be managed in Teeth_spike control code (After it activates Lara spikes death animation, it will set the camera values too)

And the same for boulder death

What leaves a simplified CalculateDeathCamera()  function only for the conditions check.